src/msgraph_core/models/page_result.py

```diff
@@ -0,0 +1,92 @@
+"""
+This module defines the PageResult class which represents a page of
+items in a paged response.
+
+The PageResult class provides methods to get and set the next link and
+the items in the page, create a PageResult from a discriminator value, set
+the value, get the field deserializers, and serialize the PageResult.
+
+Classes:
+    PageResult: Represents a page of items in a paged response.
+"""
+from typing import Any, List, Optional
+from dataclasses import dataclass
+from __future__ import annotations
+
+from kiota_abstractions.serialization.parsable import Parsable  # type: ignore
+from kiota_abstractions.serialization.serialization_writer \
+     import SerializationWriter  # type: ignore
+from kiota_abstractions.serialization.parse_node import ParseNode  # type: ignore
+from typing import TypeVar, List, Optional, Generic
+
+T = TypeVar('T')
+
+
+@dataclass
+class PageResult(Parsable, Generic[T]):
+    """
+    Represents a page of items in a paged response.
+    """
+    object_type: Optional[Any] = None
+    odata_next_link: Optional[str] = None
+    value: Optional[List[T]] = None
+
+    @staticmethod
+    def create_from_discriminator_value(parse_node: ParseNode) -> PageResult:  # pylint: disable=unused-argument
+        """
+        Creates a PageResult from a discriminator value.
+        Returns:
+            PageResult: The created PageResult.
+        """
+        impprt_statement = f"from msgraph.generated.models.{str(PageResult.object_type).lower()} \
+            import {PageResult.object_type}"
+
+        # pylint: disable=exec-used
+        exec(impprt_statement)
+        if isinstance(PageResult.object_type, str):
+            return PageResult(locals()[PageResult.object_type])
+        return PageResult()
+
```

Comment:
[refactor]: I believe there's a way to implement the iterator without having to hardcode the imports. If the response is not a `Parsable` type then default to a `PageResult`. Here are my changes.

`PageResult` class:
```py
@dataclass
class PageResult(Parsable):
    odata_next_link: Optional[str] = None
    value: Optional[List[Parsable]] = None

    @staticmethod
    def create_from_discriminator_value(parse_node: Optional[ParseNode] = None) -> PageResult:
        """
        Creates a new instance of the appropriate class based on discriminator value
        Args:
            parseNode: The parse node to use to read the discriminator value and create the object
        Returns: Attachment
        """
        if not parse_node:
            raise TypeError("parse_node cannot be null")
        return PageResult()

    def get_field_deserializers(self) -> Dict[str, Callable[[ParseNode], None]]:
        """Gets the deserialization information for this object.

        Returns:
            Dict[str, Callable[[ParseNode], None]]: The deserialization information for this
            object where each entry is a property key with its deserialization callback.
        """
        return {
            "@odata.nextLink": lambda x: setattr(self, "odata_next_link", x.get_str_value()),
            "value": lambda x: setattr(self, "value", x.get_collection_of_object_values(Parsable))
        }

    def serialize(self, writer: SerializationWriter) -> None:
        """Writes the objects properties to the current writer.

        Args:
            writer (SerializationWriter): The writer to write to.
        """
        if not writer:
            raise TypeError("Writer cannot be null")
        writer.write_str_value("@odata.nextLink", self.odata_next_link)
        writer.write_collection_of_object_values("value", self.value)
```

Then in `PageIterator` class:
```py
# in constructor
if isinstance(response, Parsable) and not constructor_callable:
            parsable_factory = type(response)
        elif constructor_callable is None:
            parsable_factory = PageResult
self.parsable_factory = parsable_factory

# fetch next method
async def fetch_next_page(self) -> dict:
        """
        Fetches the next page of items from the server.
        Returns:
            dict: The response from the server.
        Raises:
            ValueError: If the current page does not contain a next link.
            InvalidURL: If the next link URL could not be parsed.
        """

        next_link = self.current_page.odata_next_link
        if not next_link:
            raise ValueError('The response does not contain a nextLink.')
        if not next_link.startswith('http'):
            raise InvalidURL('Could not parse nextLink URL.')
        request_info = RequestInformation()
        request_info.http_method = Method.GET
        request_info.url = next_link
        request_info.headers = self.headers
        if self.request_options:
            request_info.add_request_options(*self.request_options)
        error_map: Dict[str, int] = {}
        response = await self.request_adapter.send_async(request_info, self.parsable_factory, error_map)
        return response
```